src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,155 @@
+                new TrapezoidProfile.State(getElevatorPositionMeters(), getElevatorVelocityMetersPerSeconds())
+        );
+
+        lastElevatorMotorProfileGeneration = Timer.getFPGATimestamp();
+    }
+
+    void generateAngleMotorProfile(Rotation2d targetAngle, double SpeedPercentage) {
+        angleMotorProfile = new TrapezoidProfile(
+                Conversions.scaleConstraints(ArmConstants.ANGLE_CONSTRAINTS, SpeedPercentage),
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(getAnglePosition().getDegrees(), getAngleVelocityDegreesPerSeconds())
+        );
+
+        lastAngleMotorProfileGeneration = Timer.getFPGATimestamp();
+    }
+
+    void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            masterAngleMotor.stopMotor();
+            followerAngleMotor.stopMotor();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleProfileTime());
+        setAngleMotorsVoltage(calculateAngleOutput(targetState));
+    }
+
+    void setTargetElevatorPositionFromProfile() {
+        if (angleMotorProfile == null) {
+            masterElevatorMotor.stopMotor();
+            followerElevatorMotor.stopMotor();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = elevatorMotorProfile.calculate(getElevatorProfileTime());
+        setElevatorMotorsVoltage(calculateElevatorOutput(targetState));
+    }
+
+    boolean atAngle(Rotation2d angle) {
+        return Math.abs(angle.getDegrees() - getAnglePosition().getDegrees()) < ArmConstants.ANGLE_TOLERANCE_DEGREES;
+    }
+
+    boolean atElevatorMeters(double PositionMeters) {
+        return Math.abs(PositionMeters - getElevatorPositionMeters()) < ArmConstants.ELEVATOR_TOLERANCE_DEGREES;
+    }
+
+    boolean isElevatorOpening(double targetMeters) {
+        return targetMeters > getElevatorPositionMeters();
+    }
+
+    private double getElevatorVelocityMetersPerSeconds() {
+        double elevatorVelocityMagTicksPer100ms = elevatorEncoder.getSelectedSensorVelocity();
+        double elevatorVelocityMagTicksPerSecond = Conversions.perHundredMsToPerSecond(elevatorVelocityMagTicksPer100ms);
+        double elevatorVelocityRevolutionPerSecond = Conversions.magTicksToRevolutions(elevatorVelocityMagTicksPerSecond);
+        return elevatorVelocityRevolutionPerSecond / ArmConstants.ELEVATOR_METERS_PER_REVOLUTION;
+    }
+
+    private Rotation2d getAnglePosition() {
+        double positionRevolutions = ArmConstants.ANGLE_ENCODER_POSITION_SIGNAL.refresh().getValue();
+        return Rotation2d.fromRotations(positionRevolutions);
+    }
+
+    private double getElevatorPositionMeters() {
+        double elevatorPositionMagTicks = elevatorEncoder.getSelectedSensorPosition();
+        double elevatorPositionRevolution = Conversions.magTicksToRevolutions(elevatorPositionMagTicks);
```

Comment:
elevatorPositionRevolutions

Reply:
why?
I returned meters

Reply:
Rename the variable

Reply:
but why?
I really don't understand

Reply:
It's like you said mag ticks and not mag tick.